chain/messagepool/check.go

```diff
@@ -32,14 +32,14 @@ func (mp *MessagePool) CheckMessages(ctx context.Context, protos []*api.MessageP
 // CheckPendingMessages performs a set of logical sets for all messages pending from a given actor
 func (mp *MessagePool) CheckPendingMessages(ctx context.Context, from address.Address) ([][]api.MessageCheckStatus, error) {
 	var msgs []*types.Message
-	mp.lk.Lock()
+	mp.lk.RLock()
 	mset, ok := mp.pending[from]
 	if ok {
 		for _, sm := range mset.msgs {
```

Comment:
That will undoubtly makes thoses critical sections faster (replace the `allocate, copy, grow -> repeat` pattern of append with a single allocation).
```suggestion
		msgs = make([]*types.Message, len(mset.msgs))
		for _, sm := range mset.msgs {
```

Reply:
You're right, but this method isn't on the critical path for mpool functionality. Still worth doing, perhaps as a separate patch.

Reply:
Thank you @Jorropo I have logged your change in the first item of this enhancement issue: https://github.com/filecoin-project/lotus/issues/10597